### PR TITLE
FOB Tents 

### PR DESCRIPTION
**Pull request by fira**
# About the pull request

Sprites stolen from thwomper and sammy, available NOW with game code!

Adds a few tents to be used in FOB building, mainly for organizational purposes but also providing small gameplay benefits. At current the main goal is to incentive usage to organize and liven up FOB, so the buffs are rather small.

There are 4 tent types:
 * The Command Tent is a 2x3 structure that comes bundled with an overwatch console, a phone, and two (2) chairs.
 * The Medical Tent is a 2x3 structure that comes with a NanoMED, 2 roller beds, and slightly buffs surgery (10% less time taken, and a very token pain/failure chance improvement)
 * The Requisitions Tent is a 4x3 structure that comes with a phone, rack, desks, and a variant of the old APC vendor that can stock materials and regular ammunition. The vendor starts empty, save for some tables/racks/paperwork for organization purposes. It is only useable with requisitions access.
 * The Big Tent is a bigger tent for all your organizational needs: 3x3. Get creative.

The tents also provide decent additional protection against cold environements. Unfortunately, rain/snow will visually pour through it, i can't do much about that.

The tents are extremely vulnerable to explosives and xeno claws. For simplicity and technical reasons, they are currently NON REDEPLOYABLE and NON REPLACEABLE. The tent destruction will destroy/disable linked objects (console/vendor etc). Be mindful of where you place them.

**Mind that the tents may not work out for all LZ FOBs due to the required space. I expect people will find ways to make it work anyway but it might take a while.**

# Explain why it's good for the game

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

I'm lazyyy i forgot and already closed the game... If you actually want em bug me and i'll add em
</details>

# Changelog

:cl: Firartix , Thwomper and Sammy
add: Added four types of tents to liven up FOB. They provide cold protection and benefits depending on their type. The tents spawn in Requisitions roundstart near the mortar. They're vulnerable to explosives and xenomorphs, and NON REPLACEABLE. Mind where you put them!
add: The Command tent comes equipped with an overwatch console and a phone.
add: The Medical tent provides a small boost to surgery speed/pain carried out inside it.
add: The Requisitions tent provides a restockable vendor, desk, and furniture for organization.
add: The Big tent is just a big tent, and provides you a slate to organize the way you want.
/:cl:

### CODE REVIEW

**Comment by fira** [comment]
CI C&D errors are due to instanciating a tent-linked object without a tent for tests

Not sure how to handle these

File: code/modules/tents/folded_tents.dm [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	name = "folded USCM Requisitions Tent"
+	icon_state = "req"
+	desc = "A standard USCM Requisitions Tent. Now, you can enjoy req line anywhere you go!"
+	dim_x = 4
+	dim_y = 3
+	off_x = -2
+	template = /datum/map_template/tent/reqs
+
+/obj/item/folded_tent/big
+	name = "folded USCM Medical Tent"
```
[the remark this answers is not in the document]
Comment (fira) [reply code/modules/tents/folded_tents.dm]: Yup, fixed, thanks!

**Comment by nauticall** [comment]
oi I made the original tent sprites credit me too

**Comment by GODdudesoyeah** [comment]
soul +1

**Comment by Huffie56** [comment]
image

File: code/modules/tents/folded_tents.dm [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+			if(isliving(atom) || (atom.density && atom.can_block_movement) || istype(atom, /obj/structure/tent))
+				if(message_receiver)
+					to_chat(message_receiver, SPAN_WARNING("You cannot deploy the [src] here, something ([atom.name]) is in the way."))
+				if(display_error)
+					new /obj/effect/overlay/temp/tent_deployment_error(turf)
+				return FALSE
+	return TRUE
+
+/obj/item/folded_tent/proc/unfold(turf/ref_turf)
+	var/turf/starting_turf = locate(ref_turf.x, ref_turf.y, ref_turf.z)
```
Comment (fira) [inline code/modules/tents/folded_tents.dm:48]: fix this idiot wtf is this supposed to do???

**Comment by fira** [comment]
One outstanding bug reported: when tent collapse while in, it doesn't revoke roof-seethru privileges to people so they can keep seeing inside forever (or until exiting another tent)

File: code/modules/tents/folded_tents.dm [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	return block(starting_turf, block_end_turf)
+
+/obj/item/folded_tent/attack_self(mob/living/user)
+	. = ..()
+	var/turf/deploy_turf = user.loc
+	if(!istype(deploy_turf))
+		return // In a locker or something. Get lost you already have a home.
+
+	switch(user.dir) // Fix up offset deploy location so tent is better centered + can be deployed under all angles
+		if(NORTH) deploy_turf = locate(deploy_turf.x + off_x, deploy_turf.y + 1, deploy_turf.z)
```
Comment (harryob) [inline code/modules/tents/folded_tents.dm:66]:
```suggestion
		if(NORTH) 
			 deploy_turf = locate(deploy_turf.x + off_x, deploy_turf.y + 1, deploy_turf.z)
```

File: code/modules/tents/folded_tents.dm [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+		if(SOUTH) deploy_turf = locate(deploy_turf.x + off_x, deploy_turf.y - dim_y, deploy_turf.z)
```
Comment (harryob) [inline code/modules/tents/folded_tents.dm:67]:
```suggestion
		if(SOUTH) 
			 deploy_turf = locate(deploy_turf.x + off_x, deploy_turf.y - dim_y, deploy_turf.z)
```

File: code/modules/tents/folded_tents.dm [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+		if(EAST)  deploy_turf = locate(deploy_turf.x + 1, deploy_turf.y + off_y, deploy_turf.z)
```
Comment (harryob) [inline code/modules/tents/folded_tents.dm:68]:
```suggestion
		if(EAST)  
			 deploy_turf = locate(deploy_turf.x + 1, deploy_turf.y + off_y, deploy_turf.z)
```

File: code/modules/tents/folded_tents.dm [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+		if(WEST)  deploy_turf = locate(deploy_turf.x - dim_x, deploy_turf.y + off_y, deploy_turf.z)
```
Comment (harryob) [inline code/modules/tents/folded_tents.dm:69]:
```suggestion
		if(WEST)  
			 deploy_turf = locate(deploy_turf.x - dim_x, deploy_turf.y + off_y, deploy_turf.z)
```

File: code/modules/vehicles/interior/interactable/vendors.dm [hunk 7]
```diff
@@ -382,3 +382,78 @@
 			updateUsrDialog()
 			return TRUE//We found our item, no reason to go on.
 
+
+/// Modified Restockable APC-based vendor for use by Req in the deployable tent
+/obj/structure/machinery/cm_vending/sorted/vehicle_supply/tent
+	desc =  "An automated restockable storage vendor for use in organizing FOB supplies."
```
Comment (harryob) [inline code/modules/vehicles/interior/interactable/vendors.dm:388]:
```suggestion
	desc = "An automated restockable storage vendor for use in organizing FOB supplies."
```

File: code/modules/vehicles/interior/interactable/vendors.dm [hunk 8]
```diff
@@ -382,3 +382,78 @@
...
+	req_access = list(ACCESS_MARINE_CARGO)
+	density = TRUE
+	indestructible = TRUE // Deleted with the tent instead
+	needs_power = FALSE
+
+/obj/structure/machinery/cm_vending/sorted/vehicle_supply/tent/Initialize()
+	. = ..()
+	var/obj/structure/tent/located_tent = locate(/obj/structure/tent) in src.loc
```
Comment (harryob) [inline code/modules/vehicles/interior/interactable/vendors.dm:396]:
```suggestion
	var/obj/structure/tent/located_tent = locate(/obj/structure/tent) in loc
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by fira** [comment]
This is still missing implementation of closable flaps, and possibly tent camouflages if we deem them in-scope

**Comment by fira** [comment]
Now has curtains and camouflage variants, courtesy of thwomper 

This also refactors weapon camouflage to be map_config based

**Review by harryob — CHANGES_REQUESTED** [review]

**Pull request merged**